src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -285,23 +287,25 @@ private void OnSubjectBuffersConnected(object sender, SubjectBuffersConnectedEve
             }
         }
 
-        private void UpdateReferenceLocationsTask(Task<IInlineRenameLocationSet> allRenameLocationsTask)
+        private void UpdateReferenceLocationsTask(JoinableTask<IInlineRenameLocationSet> findRenameLocationsTask)
         {
             AssertIsForeground();
 
             var asyncToken = _asyncListener.BeginAsyncOperation("UpdateReferencesTask");
-            _allRenameLocationsTask = allRenameLocationsTask;
-            allRenameLocationsTask.SafeContinueWithFromAsync(
-                async t =>
-                {
-                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationTokenSource.Token);
-                    _cancellationTokenSource.Token.ThrowIfCancellationRequested();
 
-                    RaiseSessionSpansUpdated(t.Result.Locations.ToImmutableArray());
-                },
-                _cancellationTokenSource.Token,
-                TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
-                TaskScheduler.Default).CompletesAsyncOperation(asyncToken);
+            _allRenameLocationsTask = ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+            {
+                var inlineRenameLocations = await findRenameLocationsTask.JoinAsync().ConfigureAwaitRunInline();
+
+                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationTokenSource.Token);
+                _cancellationTokenSource.Token.ThrowIfCancellationRequested();
+
+                RaiseSessionSpansUpdated(inlineRenameLocations.Locations.ToImmutableArray());
```

Comment:
Can we get a comment here saying it's unfortunate that _allRenameLocationsTask now requires the UI thread before it can let continuations run? Don't get me wrong: this is correctly expressing how it has to work right now, but it's still not how it was _supposed_ to work.